depthai_ros_driver/include/depthai_ros_driver/conversion.hpp

```diff
@@ -79,36 +96,54 @@ struct adapt_dai2ros<depthai_datatype_msgs::RawImgFrame> {
-                bridge.encoding = sensor_msgs::image_encodings::TYPE_8UC1;
-                break;
-            case CV_16UC1:
-                bridge.encoding = sensor_msgs::image_encodings::TYPE_16UC1;
-                break;
-            default:
-                ROS_WARN_STREAM_ONCE("Unknown type: " << bridge.image.type() << " in adapt_dai2ros::convert");
-                break;
-        }
         bridge.header.frame_id = frame_id + "_optical_frame";
         bridge.header.stamp.sec = input.ts.sec;
         bridge.header.stamp.nsec = input.ts.nsec;
 
-        pub.image_pub.publish(bridge.toImageMsg());
-        sensor_msgs::CameraInfo info;  // @TODO(kunaltyagi): get camera info from somewhere
-        pub.camera_info_pub.publish(info);
+        if (static_cast<dai::RawImgFrame::Type>(input.fb.type) == dai::RawImgFrame::Type::BITSTREAM) {
+            auto msg = boost::make_shared<sensor_msgs::CompressedImage>();
+            msg->header = bridge.header;
+            msg->format = "jpeg";  // @TODO: read from encoder
+            msg->data = std::move(input.data);
+            pub.compressed_image_pub.publish(msg);
+        } else {
+            bridge.image = convert_img(input);
+
+            switch (bridge.image.type()) {
+                case CV_8UC3:
+                    bridge.encoding = sensor_msgs::image_encodings::TYPE_8UC3;
+                    break;
+                case CV_8UC1:
+                    bridge.encoding = sensor_msgs::image_encodings::TYPE_8UC1;
+                    break;
+                case CV_16UC1:
+                    bridge.encoding = sensor_msgs::image_encodings::TYPE_16UC1;
+                    break;
+                default:
+                    ROS_WARN_STREAM_ONCE("Unknown type: " << bridge.image.type() << " in adapt_dai2ros::convert");
+                    break;
+            }
+            pub.raw_image_pub.publish(bridge.toImageMsg());
+        }
+
+        const auto camera_info = boost::make_shared<sensor_msgs::CameraInfo>(pub.info_manager_ptr->getCameraInfo());
+        camera_info->header = bridge.header;
+        pub.camera_info_pub.publish(camera_info);
     }
 
     static ImagePublishers create_publisher(ros::NodeHandle& nh, const std::string& name, std::size_t q_size) {
         ImagePublishers pubs;
-        pubs.image_pub =
-                nh.advertise<OutputType>(name + "/image_raw", q_size);  // @TODO(kunaltyagi): compressed if needed
+        pubs.raw_image_pub = nh.advertise<OutputType>(name + "/image_raw", q_size);
+        pubs.compressed_image_pub = nh.advertise<OutputType>(name + "/image_raw/compressed", q_size);
         pubs.camera_info_pub = nh.advertise<sensor_msgs::CameraInfo>(name + "/camera_info", q_size);
+
+        std::string camera_name = "default";
+        if (!nh.getParam("camera_name", camera_name)) {
+            nh.setParam("camera_name", camera_name);
+        }
+        const auto uri = camera_param_uri + camera_name + "/" + name + ".yaml";
```

Comment:
```suggestion
        const auto uri = camera_param_uri + "/" + camera_name + "/" + name + ".yaml";
```
I noticed this when trying to convert depth+camera_info to depth